plugins/inventory/terraform_state.py

```diff
@@ -469,32 +475,53 @@ class TerraformProviderInstance:
     types: List[str]
 
 
-ProvidersMapping = {
-    "aws": TerraformProviderInstance(provider_name="registry.terraform.io/hashicorp/aws", types=["aws_instance"]),
-    "azurerm": TerraformProviderInstance(
+PROVIDERS_CONFIG = [
+    TerraformProviderInstance(provider_name="registry.terraform.io/hashicorp/aws", types=["aws_instance"]),
+    TerraformProviderInstance(
         provider_name="registry.terraform.io/hashicorp/azurerm",
         types=["azurerm_virtual_machine", "azurerm_linux_virtual_machine", "azurerm_windows_virtual_machine"],
     ),
-    "google": TerraformProviderInstance(
+    TerraformProviderInstance(
         provider_name="registry.terraform.io/hashicorp/google", types=["google_compute_instance"]
     ),
-}
+]
+TERRAFORM_STATE_FILE_SUPPORT_VERSION = 4
 
 
-def filter_instances(
-    resources: List[TerraformModuleResource], providers: List[TerraformProviderInstance]
-) -> List[TerraformModuleResource]:
-    return [
-        r
-        for r in resources
-        if any(r.type in provider.types and r.provider_name == provider.provider_name for provider in providers)
-    ]
+def parse_provider_from_state_file_resource(provider: str) -> Optional[str]:
+    """
+    Read the provider from the State file resource.
+    Possibles format:
+      "provider[\"registry.terraform.io/hashicorp/random\"]" -> for root resources
+      "module.my-aws-module.provider[\"registry.terraform.io/hashicorp/aws\"]" -> for module resources
+    """
+    result = None
+    m = re.search('^.*provider\\["(.*)"\\]', provider)
```

Comment:
```suggestion
    m = re.search('^.*provider\["(.*)"\]', provider)
```

Reply:
You need to use a raw string: `r'^.*provider\["(.*)"\]'`